dial_test.go

```diff
@@ -338,3 +360,19 @@ func TestConn_ReadWrite(t *testing.T) {
 	assert.Nil(t, err)
 	assert.NotNil(t, resp)
 }
+
+func TestConn_ContextCancel(t *testing.T) {
```

Comment:
Could we test a cancellation in progress somehow?

Reply:
I don't see a solution for this.

But I could create a mock dialer that would endlessly try to call `TtDialer.dial` (until the first error). Then call `mockdialer.Dial` and after some timeout cancel a context. It is not a 100% that we will cancel in progress, but it is a high chance (I suppose).

Reply:
Added a new test (as described above).